PWGLF/Tasks/Resonances/chargedkstaranalysis.cxx

```diff
@@ -457,6 +491,41 @@ struct chargedkstaranalysis {
   }
   PROCESS_SWITCH(chargedkstaranalysis, processMEnew, "Process Mixed events new",
                  true);
+
+  void processMCTrue(aod::ResoMCParents const& resoParents)
+  {
+    for (const auto& part : resoParents) {                                        // loop over all pre-filtered MC particles
+      if (std::abs(part.pdgCode()) != o2::constants::physics::Pdg::kKPlusStar892) // K*892(pm)
+        continue;
+      if (std::abs(part.y()) > 0.5) // rapidity cut
+        continue;
+      bool pass1 = false;
+      bool pass2 = false;
+
+      if (part.daughterPDG1() == PDG_t::kPiPlus && part.daughterPDG2() == PDG_t::kK0Short) { // One decay to K0s and the other to pi+ (K*(892)+ mother) - Particle pass
+        pass1 = true;
+        histos1.fill(HIST("hK892pmCounter"), 0.5);
+        histos1.fill(HIST("k892pPtGen"), part.pt());
+      }
+      if (part.daughterPDG1() == PDG_t::kPiMinus && part.daughterPDG2() == -310) { // One decay to AntiK0s and the other to pi- (K*(892)- mother) - Antiparticle pass
```

Comment:
310 is [`kK0Short`](https://root.cern/doc/master/TPDGCode_8h.html#a9330fd954f94d29d85eedf5137fd3b70a4e92109c930a25e21ed582928751bd8f)
I don't think the code can eve be negative.